### PR TITLE
MP-9366 - Generate Custom Integration Event

### PR DESCRIPTION
**Pull request by dejobratic**
- Added Scriban template for IntegrationEvent;
- Added domain event handlers for CountryCreated and CountryUpdated that will raise the CountryPopulationHigherThan100M integration event (not yet fully implemented - waiting for https://github.com/NoxOrg/Nox.Generator/issues/767 to be completed)

### CODE REVIEW

**Review by rochar — APPROVED** [review]
looks good, I will approve and we can merge it

but the template is not handling the Array neither the collection =>  see the base generator to understand what is expected....

**Comment by rochar** [comment]
> * Added Scriban template for IntegrationEvent;
> * Added domain event handlers for CountryCreated and CountryUpdated that will raise the CountryPopulationHigherThan100M integration event (not yet fully implemented - waiting for [[Nox.Generators] Dispatch Domain Events before Saving to the db - MP-9237 #767](https://github.com/NoxOrg/Nox.Generator/issues/767) to be completed)

yes, lets keep the issue in progress until is complete

File: src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using {{codeGeneratorState.ApplicationNameSpace}}.Dto;
+
+namespace {{codeGeneratorState.ApplicationNameSpace}}.IntegrationEvents;
+
+{{ if integrationEvent.Description -}}
+/// <summary>
+/// {{integrationEvent.Description}}{{if !(integrationEvent.Description | string.ends_with ".")}}.{{end}}
+/// </summary>
+{{ end -}}
+public partial class {{integrationEvent.Name}} : IIntegrationEvent
```
Comment (rochar) [inline src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs]:
```suggestion
public partial class {{className}} : IIntegrationEvent
```

File: src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+{{- if integrationEvent.ObjectTypeOptions -}}
```
Comment (rochar) [inline src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs:20]: we will need also the other aditional cases for CollectionTypeOptions and ArrayTypeOptions

we can do this in the next PR, so we are not to far away

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs (hunk 2 above)
Comment (dejobratic) [reply src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs:20]: Thanks for the feedback. Will add that in the next PR.

File: src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs (hunk 1 above)
Comment (dejobratic) [reply src/Nox.Generator/Application/IntegrationEvents/IntegrationEvent.template.cs]: Changes were applied. Thanks.

**Review by rochar — APPROVED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**